### PR TITLE
Added geospatial tests for $nearSphere

### PR DESCRIPTION
**Pull request by vic-tsang**
This PR contains:
- geospatial tests for $nearSphere

Ref:
- Issue: #32

### CODE REVIEW

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_edge_cases.py [hunk 1]
```diff
@@ -0,0 +1,183 @@
+"""Tests for $nearSphere edge cases — distance calculation, sort, limit/skip."""
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_edge_cases.py]: Sort tests (SORT_AND_DISTANCE_TESTS and test_nearSphere_explicit_sort_overrides_distance) belong in test_nearSphere_core_functionality.py — distance-based sorting is the defining behavior of $nearSphere, not an edge case.

Both docstrings are also inaccurate:
- test_nearSphere_core_functionality.py claims "sorting, null/missing" but contains neither
- This file claims "limit/skip" but contains none

Please move the sort tests and update both docstrings to match actual contents.

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py [hunk 2]
```diff
@@ -0,0 +1,390 @@
+"""Tests for $nearSphere error handling — all error/rejection cases consolidated.
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py]: $minDistance is entirely missing from this PR. Per the MongoDB docs (https://www.mongodb.com/docs/manual/reference/operator/query/nearSphere/), it's a first-class parameter — the primary documented example is a ring query with both $minDistance and $maxDistance. Please add:

- $minDistance success: GeoJSON (meters), legacy + 2dsphere (radians), ring query
- $minDistance with 2d index (docs say "available only if the query uses the 2dsphere index")
- $minDistance error cases (negative, non-numeric)

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py]: Missing $maxDistance error cases. It's used in many success tests but never validated with invalid inputs. Add:

- Negative value
- NaN / Infinity
- Non-numeric types (string, bool, object, array, null)

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+)
+from documentdb_tests.framework.executor import execute_command
+from documentdb_tests.framework.parametrize import pytest_params
+from documentdb_tests.framework.test_constants import (
+    FLOAT_INFINITY,
+    FLOAT_NAN,
+    FLOAT_NEGATIVE_INFINITY,
+)
+
+GEOJSON_TYPE_VALIDATION_TESTS: list[QueryTestCase] = [
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py:22]: This covers LineString, Polygon, MultiPoint but the MongoDB docs say "validates that the type is Point, any other input type returns an error." Missing:

- "MultiLineString", "MultiPolygon", "GeometryCollection"
- Unknown string (e.g., "InvalidType")
- Non-string type: {"type": 123, "coordinates": [0, 0]}
- Null: {"type": null, "coordinates": [0, 0]}
- Missing type field entirely: {"coordinates": [0, 0]}

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+        msg="Should reject $nearSphere inside $nor",
+    ),
+]
+
+
+@pytest.mark.usefixtures("geo_2dsphere")
+@pytest.mark.parametrize("test", pytest_params(OPERATOR_RESTRICTION_TESTS))
+def test_nearSphere_operator_restriction_errors(collection, test):
+    """Verifies $nearSphere rejects invalid operator combinations."""
+    collection.create_index([("loc2", "2dsphere")])
```
Comment (fanyangv) [inline documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py:375]: The test case two_nearSphere_different_fields queries on loc1 and loc2, but only loc2 gets an index here (loc comes from the fixture — not loc1). This means loc1 has no geo index.

This passes only because the "can't have two $nearSphere" error fires at parse time before index lookup. If an engine validates at planning time, you'd get NO_QUERY_EXECUTION_PLANS_ERROR instead.

Either add collection.create_index([("loc1", "2dsphere")]), or change the test case to use loc (which already has an index from the fixture) instead of loc1.

**Review by fanyangv — CHANGES_REQUESTED** [review]

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py (hunk 2 above)
Comment (fanyangv) [reply documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py]: I see $minDistance is covered in PR #226 (specifiers/minDistance/) including error cases, BSON validation, and legacy mode. PR #224 also includes $nearSphere-specific ring query tests (geojson_with_minDistance_annular). Ignore this comment.

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py (hunk 3 above)
Comment (fanyangv) [reply documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py]: I see these are covered in PR #224 (specifiers/maxDistance/test_maxDistance_errors.py) including negative, NaN, Infinity, non-numeric types, and invalid context. Ignore this comment.

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_edge_cases.py (hunk 1 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_edge_cases.py]: moved and updated doc string

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py (hunk 2 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py]: ok

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py (hunk 3 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py]: ok

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py (hunk 4 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py:22]: added except for Missing type field entirely it succeeds. Added as 'missing_type_defaults_to_point' in test_nearSPhere_argument_handling.py

File: documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py (hunk 5 above)
Comment (vic-tsang) [reply documentdb_tests/compatibility/tests/core/operator/query/geospatial/nearSphere/test_nearSphere_errors.py:375]: changed to use loc instead of loc1

**Review by fanyangv — APPROVED** [review]

**Review by eerxuan — COMMENTED** [review]
Apart from feedback from https://github.com/documentdb/functional-tests/pull/218#discussion_r3313542027, LGTM.

**Review by eerxuan — APPROVED** [review]
